compiler/src/dmd/mtype.d

```diff
@@ -4783,6 +4783,12 @@ extern (C++) final class TypeFunction : TypeNext
                 else if (pMessage && !*pMessage)
                     *pMessage = getParamError(args[u], p);
 
+                if (argLoc && argumentList.arguments.length >= u)
```

Comment:
The bounds check and indexing are not right. On line 4751 there's:

```D
if (u < args.length)
{
    Expression arg = args[u];
```

Move the `arg` variable outside the block and you can use it for `argLoc`.